source/features/scroll-to-current-file.tsx

```diff
@@ -0,0 +1,57 @@
+import * as pageDetect from 'github-url-detection';
+
+import features from '../feature-manager.js';
+import GitHubFileURL from '../github-helpers/github-file-url.js';
+
+function scrollToCurrentFile(): void {
+	const url = new GitHubFileURL(location.href);
+	const filePath = url.filePath;
+
+	// we need to escape the filePath if we use querySelector.
+	// eslint-disable-next-line unicorn/prefer-query-selector
+	const item = document.getElementById(`${filePath}-item`);
+	const text = item?.querySelector('.PRIVATE_TreeView-item-content-text');
+	text?.scrollIntoView({block: 'center', behavior: 'auto'});
```

Comment:
I think you just need this:

```suggestion
	scrollIntoViewIfNeeded(document.getElementById(`${filePath}-item`));
```

We have a `scrollIntoViewIfNeeded` function

Reply:
The Directory item wraps the entire child items, so it needed to select the first text child.
Thanks for the `scrollIntoViewIfNeeded` function.

Reply:
Okay. I was trying to center the directory but it doesn't seem necessary. I'll change that.